src/Owin.Security.Providers.Salesforce/SalesforceAuthenticationHandler.cs

```diff
@@ -187,7 +187,12 @@ protected override Task ApplyResponseChallengeAsync()
             var state = Options.StateDataFormat.Protect(properties);
 
             var authorizationEndpoint =
-                $"{Options.Endpoints.AuthorizationEndpoint}?response_type={"code"}&client_id={Options.ClientId}&redirect_uri={HttpUtility.UrlEncode(redirectUri)}&display={"page"}&immediate={false}&state={Uri.EscapeDataString(state)}&scope={""}";
+                $"{Options.Endpoints.AuthorizationEndpoint}?response_type={"code"}&client_id={Options.ClientId}&redirect_uri={HttpUtility.UrlEncode(redirectUri)}&display={"page"}&immediate={false}&state={Uri.EscapeDataString(state)}";
+
+            if (Options.Scope.Count > 0)
```

Comment:
mind using the search safe operator to make sure its not null? 

`Options?.Scope?.Count > 0 ?? false`